src/com/goide/psi/impl/GoPsiImplUtil.java

```diff
@@ -1131,4 +1131,19 @@ public static boolean isUnaryBitAndExpression(@Nullable PsiElement parent) {
     PsiElement grandParent = parent != null ? parent.getParent() : null;
     return grandParent instanceof GoUnaryExpr && ((GoUnaryExpr)grandParent).getBitAnd() != null;
   }
+
+  @Nullable
+  public static PsiElement getBreakStatementOwner(@NotNull GoBreakStatement breakStmt) {
+    GoLabelRef label = breakStmt.getLabelRef();
+    if (label != null) {
+      return label.getReference().resolve();
+    }
+    GoCompositeElement breaksOutOf =
+      PsiTreeUtil.getParentOfType(breakStmt, GoSwitchStatement.class, GoForStatement.class, GoSelectStatement.class,
+                                  GoFunctionOrMethodDeclaration.class);
+    if (breaksOutOf instanceof GoFunctionOrMethodDeclaration) {
```

Comment:
I think we don't need to stop on GoFunctionOrMethodDeclaration. Only in optimisation purposes, but two getParent invocations doesn't matter much.